### DISCUSSION TITLE
Custom "Edit" forms with validation

### DISCUSSION THREAD

**Discussion by PieterL75** [netbox-community/netbox#16282] [Ideas]
NetBox is very nice in all the fields that exist and relate to eachother across the models.
You can clearly see that when editing a device for example.

And there comes my pain.. all those fields, most of them are not used in my company. 
I have people asking why there are coordinates to be entered for a server in a rack..

I would love to see 'customizable' edit forms, that can hide fields, and add extra validations to those fields.
It could even be, that the form is different for a different device role/model, ...

It is a rough idea, but it would make the UI more clear to the users.

Any suggestions/thoughts ?

Pieter

**Comment by jeremystretch** [comment]
This has been proposed in the past and rejected, as it would introduce a substantial development and maintenance burden in exchange for very little value. If you absolutely need to hide certain fields for whatever reason, you'll need to fork the code base and make the modifications locally.

**Reply by PieterL75** [reply]
Nice to hear that this has been proposed before.
I do agree on the fact that this will require an amount of effort, but it would create one very nice GUI that fits everybody's needs.
Forking is indeed an option, but increases maintenance and possible breakage in the future releases...

Maybe the grand feature for 5.0 :-)
